dao/src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.specification.Specification;
+import com.epam.esm.entity.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class TagDaoImpl implements TagDao {
+    private final JdbcTemplate jdbcTemplate;
+
+    private final String FIND_BY_ID = """
+            SELECT id, name FROM tags WHERE id=? AND is_deleted=false
+            """;
```

Comment:
It's good practice to use string blocks if your string value is too long. Your string contains only 1 line, and it's rather short.
Please, don't use string blocks in such cases

Reply:
Fixed